addons/mrp/mrp.py

```diff
@@ -1236,8 +1238,7 @@ def action_confirm(self, cr, uid, ids, context=None):
         """
         user_lang = self.pool.get('res.users').browse(cr, uid, [uid]).partner_id.lang
         context = dict(context, lang=user_lang)
-        uncompute_ids = filter(lambda x: x, [not x.product_lines and x.id or False for x in self.browse(cr, uid, ids, context=context)])
-        self.action_compute(cr, uid, uncompute_ids, context=context)
+        self.action_compute(cr, uid, ids, context=context)
```

Comment:
This change is not going to keep the modification you done in BoM line.
For example, after BoM lines' been computed, if you replace any product from one of that line then that change will be reverted on validation of the MO.